qiskit_metal/toolbox_metal/layer_stack_handler.py

```diff
@@ -0,0 +1,439 @@
+
+            except Exception as ex:
+                self._warning_search(chip_name, layer_num, datatype, ex)
+
+        return found_fill_value, fill_value
+
+    def get_z_coord_for_chip_and_layer(self,
+                                       chip_name: str,
+                                       layer_number: int,
+                                       datatype: int = 0) -> float:
+        """Using the layer_stack information, return the z_coord.
+
+        Args:
+            chip_name (str): Name of chip which has the layer of interest.
+            layer_number (int): The unique layer number through out all chips.
+                            The same layer number can  not be used on multiple chips.
+            datatype (int, optional): Datatype of the layer number. Defaults to 0.
+
+        Returns:
+           float: return the z_coord or 0.0 if not found in the dataframe.
+        """
+        z_coord = 0.0
+        mask = (self.ls_df['layer']
+                == layer_number) & (self.ls_df['datatype'] == datatype) & (
+                    self.ls_df['chip_name'].str.contains(chip_name))
+        search_result_df = self.ls_df[mask]
+        if len(search_result_df) > 0:
+            try:
+                #Use the parse_values from Metal when parsing the values from layer_stack.
+
+                z_coord = self.multi_planar_design.parse_value(
+                    search_result_df.z_coord.iloc[0].strip('\''))
+            except Exception as ex:
+                self._warning_search(chip_name, layer_number, datatype, ex)
+
+        return z_coord
+
+    def get_thickness_for_chip_and_layer(self,
+                                         chip_name: str,
+                                         layer_number: int,
+                                         datatype: int = 0) -> float:
+        """Using the layer_stack information, return the thickness.
+
+        Args:
+            chip_name (str): Name of chip which has the layer of interest.
+            layer_number (int): The unique layer number through out all chips.
+                            The same layer number can  not be used on multiple chips.
+            datatype (int, optional): Datatype of the layer number. Defaults to 0.
+        Returns:
+           float: return the thickness or 0.0 if not found in the dataframe.
+        """
+        thickness = 0.0
+        mask = (self.ls_df['layer']
+                == layer_number) & (self.ls_df['datatype'] == datatype) & (
+                    self.ls_df['chip_name'].str.contains(chip_name))
+        search_result_df = self.ls_df[mask]
+
+        if len(search_result_df) > 0:
+            try:
+                thickness = parse_units(
+                    search_result_df.thickness.iloc[0].strip('\''))
+            except Exception as ex:
+                self._warning_search(chip_name, layer_number, datatype, ex)
+
+        return thickness
+
+    def get_thickness_and_zcoord_for_chip_and_layer(
```

Comment:
Is this function explicitly needed with the same logic? The work of this function is already done by `get_thickness_for_chip_and_layer` and `get_zcoord_for_chip_and_layer`.

Reply:
Would it be better to have just a single function where user can pass in different params that they want, and it returns all at once? Something with the following signature:

```py
def get_properties_for_chip_and_layer(self, properties: List[str], chip_name: str, layer_number: int, datatype: int)
```

Reply:
Agreed.  Since the file design was changing, I had made unique method, by since the file format has settled down, we should do it this way.

Reply:
I committed a method for this.